x/staking/side_handler_test.go

```diff
@@ -823,12 +824,17 @@ func (suite *SideHandlerTestSuite) TestSideHandleMsgSignerUpdate() {
 		txReceipt := &ethTypes.Receipt{BlockNumber: blockNumber}
 		suite.contractCaller.On("GetConfirmedTxReceipt", msgTxHash.EthHash(), chainParams.MainchainTxConfirmations).Return(txReceipt, nil)
 
+		// uncompressed the pub key for staking event
+		uncompressed, err := ethcrypto.DecompressPubkey(pubKey)
+		require.NoError(t, err)
+		uncompressedBytes := ethcrypto.FromECDSAPub(uncompressed)
+
 		signerUpdateEvent := &stakinginfo.StakinginfoSignerChange{
 			ValidatorId:  new(big.Int).SetUint64(oldSigner.ID.Uint64()),
 			Nonce:        nonce,
 			OldSigner:    hmCommonTypes.HexToHeimdallAddress(oldSigner.Signer).EthAddress(),
 			NewSigner:    hmCommonTypes.HexToHeimdallAddress(newSigner[0].Signer).EthAddress(),
-			SignerPubkey: hmCommonTypes.NewPubKeyFromHex(newSigner[0].PubKey).Bytes()[1:],
+			SignerPubkey: uncompressedBytes,
```

Comment:
Event from contracts has uncompressed pub key without 1-byte prefix. Testcase should work without any other change in code because `helper.CompressPubKey` handles prefix (adds prefix if the prefix is not there) https://github.com/maticnetwork/heimdall/blob/master/helper/util.go#L111-L113

```suggestion
			SignerPubkey: uncompressedBytes[1:],
```